TypeSystem/flow-analysis/type_of_interest_A03_t04.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that at type `T` is a type of interest for a variable `x`
+/// in a set of tested types `tested` if `tested` contains a type `S` such that
+/// `T` is `S`, or `T` is `NonNull(S)`.
+///
+/// @description Checks that a type `Null` cannot be made a type of interest
+/// for a variable.
+/// @author sgrekhov22@gmail.com
+
+import 'dart:async';
+
+extension on Null {
+  void foo() {}
+}
+
+test1() {
+  String s = "a";
+  if (s is Null) {}  // Try to make `Null` a type of interest
+  s = "b";
+  s.foo();
```

Comment:
This is the place where a member invocation won't suffice (because `Null` does not have any members that aren't shared by _all_ objects).

But `expectStaticType` should work.